### ISSUE TITLE
is the semantic loss right?

### ISSUE THREAD

**Issue by codingwolfman** [EasonApolo/mstn#4]
# Moving Centroid
        decay = self.decay
        s_centroid = (1-decay) * self.s_centroid + decay * current_s_centroid
        t_centroid = (1-decay) * self.t_centroid + decay * current_t_centroid
        semantic_loss = self.MSEloss(s_centroid, t_centroid)
        self.s_centroid = s_centroid.detach()
        self.t_centroid = t_centroid.detach()

by this way, only the gradient of current_centroid will be backward, because the global centroid is detached, if this, we can compute semantic loss just using MSELoss(current_s_centroid, current_t_centroid), i'm not sure

**Comment by EasonApolo** [comment]
Hi, since self.s_centroid is the detached version of the last centroid, we can think of it as just a normal tensor that does not require grad. Thus its value will affect the grad of the current_s_centroid. The necessity of 'detach' is that the computing graph will continuously grow as we save last centroid and use it in the next epoch.
